### ISSUE TITLE
Runtime error on HistServ play command

### ISSUE THREAD

**Issue by ivan-avalos** [oragono/oragono#1244]
* **Version:** v2.2.0
* **OS:** Linux x86_64 (Amazon Linux 2)

I just installed oragono in my VPS. I read the manual, set and modified the `ircd.yml`, created the DB, and everything else seems to be working just fine, except for the `/history` command, which crashes the server for some reason with the following backtrace:

```
2020-08-23T21:15:14.625Z : error : internal   : Client caused panic: runtime error: invalid memory address or nil pointer dereference
goroutine 43 [running]:
runtime/debug.Stack(0xc000144ca0, 0x9f8ba0, 0xf9a6d0)
	runtime/debug/stack.go:24 +0x9d
github.com/oragono/oragono/irc.(*Client).run.func1(0xc0001daa80, 0xc0000c0b00)
	github.com/oragono/oragono/irc/client.go:592 +0xa7
panic(0x9f8ba0, 0xf9a6d0)
	runtime/panic.go:969 +0x166
database/sql.(*DB).conn(0x0, 0xb96b60, 0xc0001d6000, 0x1, 0xc000145070, 0x4e847a, 0xc0002b6040)
	database/sql/sql.go:1134 +0x41
database/sql.(*DB).query(0x0, 0xb96b60, 0xc0001d6000, 0xc000304410, 0xcf, 0xc0001453a0, 0x4, 0x6, 0xc000145101, 0x4ed5d2, ...)
	database/sql/sql.go:1552 +0x66
database/sql.(*DB).QueryContext(0x0, 0xb96b60, 0xc0001d6000, 0xc000304410, 0xcf, 0xc0001453a0, 0x4, 0x6, 0x1000000004751b2, 0x0, ...)
	database/sql/sql.go:1534 +0xd1
github.com/oragono/oragono/irc/mysql.(*MySQL).selectItems(0xc000156648, 0xb96b60, 0xc0001d6000, 0xc000304410, 0xcf, 0xc0001453a0, 0x4, 0x6, 0x0, 0x0, ...)
	github.com/oragono/oragono/irc/mysql/history.go:771 +0xd1
github.com/oragono/oragono/irc/mysql.(*MySQL).betweenTimestamps(0xc000156648, 0xb96b60, 0xc0001d6000, 0xc000023360, 0x8, 0x0, 0x0, 0x2546134e, 0xed6d4d362, 0x0, ...)
	github.com/oragono/oragono/irc/mysql/history.go:831 +0x578
github.com/oragono/oragono/irc/mysql.(*mySQLHistorySequence).Between(0xc000474780, 0x0, 0x0, 0x2546134e, 0xed6d4d362, 0x0, 0x0, 0x0, 0x2546134e, 0xed6d4c552, ...)
	github.com/oragono/oragono/irc/mysql/history.go:874 +0x1c4
github.com/oragono/oragono/irc.easySelectHistory(0xc000156400, 0xc0001daa80, 0xc000474750, 0x2, 0x3, 0x0, 0x0, 0x16001456c8, 0x2000, 0x7fa712543188, ...)
	github.com/oragono/oragono/irc/histserv.go:241 +0x3f1
github.com/oragono/oragono/irc.histservPlayHandler(0xc000156400, 0xc0001daa80, 0xc000198000, 0x4, 0xc000474750, 0x2, 0x3, 0xc0000b2100)
	github.com/oragono/oragono/irc/histserv.go:175 +0x77
github.com/oragono/oragono/irc.serviceRunCommand(0xc000020d80, 0xc000156400, 0xc0001daa80, 0xc0000aab40, 0xc000198000, 0x4, 0xc000474750, 0x2, 0x3, 0xc0000b2100)
	github.com/oragono/oragono/irc/services.go:215 +0x694
github.com/oragono/oragono/irc.serviceCmdHandler(0xc000156400, 0xc0001daa80, 0x0, 0x0, 0xc000300000, 0x8, 0xc000474740, 0x3, 0x4, 0x0, ...)
	github.com/oragono/oragono/irc/services.go:128 +0x1b8
github.com/oragono/oragono/irc.(*Command).Run.func1(0xc0000b2100, 0xc0001daa80, 0xc000145d58, 0xc000156400, 0x0, 0x0, 0xc000300000, 0x8, 0xc000474740, 0x3, ...)
	github.com/oragono/oragono/irc/commands.go:53 +0x145
github.com/oragono/oragono/irc.(*Command).Run(0xc000145d58, 0xc000156400, 0xc0001daa80, 0xc0000c0b00, 0x0, 0x0, 0xc000300000, 0x8, 0xc000474740, 0x3, ...)
	github.com/oragono/oragono/irc/commands.go:54 +0x167
github.com/oragono/oragono/irc.(*Client).run(0xc0001daa80, 0xc0000c0b00)
	github.com/oragono/oragono/irc/client.go:687 +0x526
github.com/oragono/oragono/irc.(*Server).RunClient(0xc000156400, 0xb97ea0, 0xc00029fbb0)
	github.com/oragono/oragono/irc/client.go:364 +0x97b
created by github.com/oragono/oragono/irc.(*NetListener).serve
	github.com/oragono/oragono/irc/listeners.go:99 +0x234
```

**Comment by slingamn** [comment]
Initial findings: this doesn't appear to be a full server crash, since it's correctly handled by `recover-from-errors`.

I have no official record of this for the 2.2.0 release, but `strings` output seems to indicate that 2.2.0 was compiled with go1.14.6. (The problem of officially recording/documenting this was fixed in #1234.) At first glance, it looks like the database object isn't being initialized correctly?

https://github.com/golang/go/blob/edfd6f28486017dcb136cd3f3ec252706d4b326e/src/database/sql/sql.go#L1134

**Comment by slingamn** [comment]
This looks like a config validation issue: you can set `history.persistent.enabled` without actually configuring a MySQL server, which will cause this panic.
